Classes/Finishers/RedirectFinisher.php

```diff
@@ -65,7 +65,11 @@ public function executeInternal()
 
         $uriParts = parse_url($uri);
         if (!isset($uriParts['scheme']) || $uriParts['scheme'] === '') {
-            $uri = $request->getHttpRequest()->getAttribute(ServerRequestAttributes::BASE_URI) . $uri;
+            $baseUri = $request->getHttpRequest()->getAttribute(ServerRequestAttributes::BASE_URI);
+            if (substr($baseUri, -1) === '/') {
```

Comment:
I would suggest the following. 

```php
$baseUri = $request->getHttpRequest()->getAttribute(ServerRequestAttributes::BASE_URI);
$uri = rtrim($baseUri, '/') . '/' . ltrim($uri, '/');
```

Reply:
Mmm, actually this server request attribute doesn't even exist anymore, need to doubl echeck what is going on here.